testing/run-engine-phpunit.sh

```
@@ -0,0 +1,134 @@
+#
+# Requires: Docker, and the host PHP built with `pdo_mysql` / `pdo_pgsql`.
+# If a driver extension is missing the corresponding engine is skipped with a
+# loud warning (you're then trusting CI for that engine) — it is NOT a hard
+# failure, because not every dev box has both drivers. A check that actually
+# RAN and FAILED is a hard failure.
+#
+# Usage: bash testing/run-engine-phpunit.sh
+# Part of the Local gate — see docs/internal/test-suites.md § "Local gate".
+# ---------------------------------------------------------------------------
+set -uo pipefail
+cd "$(dirname "$0")/.."
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)